lib/checks/checks.go

```diff
@@ -371,11 +373,40 @@ func (r *checker) Check(ctx context.Context) (failed []*agentpb.Probe) {
 		return nil
 	}
 
-	// check each server against its profile
+	serverC := make(chan Server, len(r.Servers))
 	for _, server := range r.Servers {
-		failed = append(failed, r.CheckNode(ctx, server)...)
+		serverC <- server
 	}
 
+	close(serverC)
+
+	// Make the number of parallel routines relative to the number of CPU cores. This allows the parallel execution
+	// to scale relative to the size of the server. Using NumCPU/2 is a guess that isn't expected to generate too much
+	// load on the server running the checks.
+	numRoutines := (runtime.NumCPU() / 2) + 1
+
+	var mutex sync.Mutex
+
+	var wg sync.WaitGroup
+
+	wg.Add(numRoutines)
+
+	for i := 0; i < numRoutines; i++ {
+		go func() {
+			for server := range serverC {
+				f := r.CheckNode(ctx, server)
+
+				mutex.Lock()
+				failed = append(failed, f...)
+				mutex.Unlock()
```

Comment:
Mixing new & old style concurrency feels weird here.  Elsewhere (e.g. executeSubphasesConcurrently) you use a channel to aggregate errors.  I don't think it needs to be fixed because the logic isn't wrong -- just something that my brain flagged while reading the code.

Reply:
Yea, I see the conundrum. I'm really not hugely tied to one approach or the other tbh, and this approach was tested so will probably just leave it for now.